core/src/http.rs

```diff
@@ -106,16 +106,78 @@ impl Display for HttpErrorResponse {
 /// value. Returns `None` if there is no `authorization` header, and an error if there is an
 /// `authorization` header whose value is not a bearer token.
 pub fn extract_bearer_token(conn: &Conn) -> Result<Option<AuthenticationToken>, anyhow::Error> {
-    if let Some(authorization_value) = conn.request_headers().get("authorization") {
-        if let Some(received_token) = authorization_value.to_string().strip_prefix("Bearer ") {
-            return Ok(Some(
-                AuthenticationToken::new_bearer_token_from_string(received_token)
-                    .context("invalid bearer token")?,
-            ));
-        } else {
+    if let Some(authorization) = conn
+        .request_headers()
+        .get("authorization")
+        .map(HeaderValue::to_string)
+    {
+        let (bearer, token) = authorization
+            .split_once(char::is_whitespace)
+            .ok_or_else(|| anyhow!("invalid bearer token"))?;
+
+        if bearer.to_lowercase() != "bearer" {
             return Err(anyhow!("authorization header value is not a bearer token"));
         }
+
+        return Ok(Some(AuthenticationToken::new_bearer_token_from_string(
+            token.trim_start(),
+        )?));
     }
 
     Ok(None)
 }
+
+#[cfg(test)]
+mod tests {
+    use assert_matches::assert_matches;
+    use trillium_testing::TestConn;
+
+    use super::extract_bearer_token;
+
+    #[test]
+    fn authorization_header() {
+        assert_matches!(
+            extract_bearer_token(
+                &TestConn::build("get", "/", "body")
+                    .with_request_header("authorization", "bearer gVfRUu9krhxrUgFsEo-P5w"),
+            ),
+            Ok(Some(_))
+        );
+        assert_matches!(
+            extract_bearer_token(
+                &TestConn::build("get", "/", "body")
+                    .with_request_header("authorization", "BeArEr     gVfRUu9krhxrUgFsEo-P5w"),
+            ),
+            Ok(Some(_))
```

Comment:
I think it'd be nice to check the contents of the `AuthenticationToken` to make sure nothing bad happens when we call `String::trim_start`.